BlazorCore/DSD.MSS.Blazor.Components.Core/Components/RequestCommentComponent.razor.cs

```diff
@@ -0,0 +1,138 @@
+﻿namespace DSD.MSS.Blazor.Components.Core.Components
+{
+
+using DSD.MSS.Blazor.Components.Core.Models;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Forms;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+
+
+
+    public partial class RequestCommentComponent
+    {
+        [Parameter]
+        public List<RequestCommentInfo> WorkComments
+        {
+            get => workComments;
+            set
+            {
+                if (workComments == value) return;
+                this.workComments = value;
+                WorkCommentsChanged.InvokeAsync(value);
+            }
+        }
+        [Parameter]
+        public bool ReadOnly { get; set; }
+        [Parameter]
+        public EventCallback<string> SubmitComment { get; set; }
+        [Parameter]
+        public EventCallback<List<RequestCommentInfo>> WorkCommentsChanged { get; set; }
+        
+        [Parameter]
+        public CommentTextModel TextModel { get; set; }
+
+        public int MaxShownComments { get; set; } = 5;
+
+        private SortOption SortOption { get; set; } = new SortOption();
+
+        public EditContext EditContext { get; set; }
+
+        private List<RequestCommentInfo> workComments { get; set; }
+
+        private CommentDTO requestComment { get; set; } = new CommentDTO();
+
+
+        private bool areAllCommentsShown { get; set; } = false;
+
+        public static List<SelectListItem> SortOptions { get; set; }
+
+
+
+        protected override void OnInitialized()
+        {
+
+            if(this.WorkComments == null)
+            {
+                this.WorkComments = new List<RequestCommentInfo>();
+                this.setMaxToCurrentCount();
+            }
+            if (this.WorkComments.Any())
+            {
+                this.WorkComments = WorkComments.OrderByDescending(o => o.CreatedDateUTC).ToList();
+                this.setMaxToCurrentCount();
+            }
+            else
+            {
+                this.areAllCommentsShown = true;
+            }
+          
+            if(this.TextModel == null)
+            {
+                this.TextModel = new CommentTextModel()
+                {
+                    AddCommentBtn = "Add Comment",
+                    CommentsLabel = "Comments",
+                    NoCommentsLabel = "No Comments",
+                    LoadMoreCommentBtn = "Load More Comments",
+                    SortOrderNew = "Newest",
+                    SortOrderOld = "Oldest"
+                };
+            }
+            SortOptions = new List<SelectListItem>() {
+            new SelectListItem { Id = "1", Text = TextModel.SortOrderNew },
+            new SelectListItem { Id = "2", Text = TextModel.SortOrderOld }
```

Comment:
Create variables for "1" and "2"  to explicitly explain what they represent. Also, do we want to allow whoever uses this component to set their own filtering methods? If so we will have to refactor the way this is used.